astropy/utils/iers/tests/test_iers.py

```diff
@@ -34,14 +34,14 @@ def test_simple(self):
         iers_tab = iers.IERS.open()
         assert iers.IERS.iers_table is not None
         assert isinstance(iers.IERS.iers_table, QTable)
-        assert iers_tab['UT1_UTC'].unit is u.second
-        assert iers_tab['PM_x'].unit is u.arcsecond
-        assert iers_tab['PM_y'].unit is u.arcsecond
+        assert (iers_tab['UT1_UTC'].unit / u.second).is_unity()
```

Comment:
Humm, this is slightly disconcerting.  I see the nice commit description that points to why this change was made.  For example:
```
In [5]: u.s is u.Unit('s')
Out[5]: True

In [6]: u.s is u.Unit('s', format='cds')
Out[6]: False

In [7]: u.s / u.s
Out[7]: Unit(dimensionless)

In [8]: u.s / u.Unit('s', format='cds')
Out[8]: Unit("s / s")
```
I had imagined that the unit `format` implies how to parse the unit string (or combinations thereof), but that the fundamental units coming out were the same thing.  Not so apparently?

So this does imply some risk of breaking code with this PR.  Thoughts @mhvk?